game/script-nicknames.rpy

```diff
@@ -0,0 +1,689 @@
+        "rectum",
+        "retard",
+        "rimjaw",
+        "rimming",
+        "s hit",
+        "s.o.b.",
+        "s_h_i_t",
+        "sadist",
+        "schlong",
+        "screwing",
+        "scroat",
+        "scrote",
+        "scrotum",
+        "semen",
+        "sex",
+        "sh!+",
+        "sh!t",
+        "sh1t",
+        "shag",
+        "shagger",
+        "shaggin",
+        "shagging",
+        "shemale",
+        "shi+",
+        "shit",
+        "shitdick",
+        "shite",
+        "shited",
+        "shitey",
+        "shitfuck",
+        "shitfull",
+        "shithead",
+        "shiting",
+        "shitings",
+        "shits",
+        "shitted",
+        "shitter",
+        "shitters",
+        "shitting",
+        "shittings",
+        "shitty",
+        "skank",
+        "slut",
+        "sluts",
+        "smegma",
+        "smut",
+        "snatch",
+        "son-of-a-bitch",
+        "spac",
+        "spunk",
+        "t1tt1e5",
+        "t1tties",
+        "teets",
+        "teez",
+        "testical",
+        "testicle",
+        "tit",
+        "titfuck",
+        "tits",
+        "titt",
+        "tittie5",
+        "tittiefucker",
+        "titties",
+        "tittyfuck",
+        "tittywank",
+        "titwank",
+        "tosser",
+        "turd",
+        "tw4t",
+        "twat",
+        "twathead",
+        "twatty",
+        "twunt",
+        "twunter",
+        "v14gra",
+        "v1gra",
+        "vagina",
+        "viagra",
+        "vulva",
+        "w00se",
+        "wang",
+        "wank",
+        "wanker",
+        "wanky",
+        "whoar",
+        "whore",
+        "willies",
+        "willy",
+        "xrated",
+        "xxx"
+    ]
+
+    # Natsuki finds these nicknames funny
+    NICKNAME_FUNNY_LIST = [
+        "gorgeous",
+        "hot",
+        "hotstuff",
+        "hottie",
+        "mama",
+        "mom",
+        "mommy",
+        "mother",
+        "mum",
+        "mummy",
+        "sexy",
+        "smol",
+        "snack"
+    ]
+
+    """
+    Returns the nickname type for a given string nickname, defaulting to NICKNAME_TYPE_NEUTRAL
+
+    IN:
+        nickname - The nickname to test
+    OUT:
+        Nickname type, integer as defined in constant list
+    """
+    def get_nickname_type(nickname):
+
+        if not isinstance(nickname, basestring):
+            return NICKNAME_TYPE_INVALID
+        
+        else:
+            nickname = nickname.lower().replace(" ", "")
+
+            if nickname in NICKNAME_LOVED_LIST:
+                return NICKNAME_TYPE_LOVED
+
+            elif nickname in NICKNAME_DISLIKED_LIST:
+                return NICKNAME_TYPE_DISLIKED
+
+            elif nickname in NICKNAME_HATED_LIST:
+                return NICKNAME_TYPE_HATED
+
+            elif nickname in NICKNAME_PROFANITY_LIST:
+                return NICKNAME_TYPE_PROFANITY
+
+            elif nickname in NICKNAME_FUNNY_LIST:
+                return NICKNAME_TYPE_FUNNY
```

Comment:
Considering this, can we change the lists to sets so we can have faster accessors? It's a minor performance improvement over `in` checks like this since these need to iterate 'till they find the item or reach the end.